src/selectors/interface.js

```diff
@@ -109,48 +110,73 @@ export const makeGetOrdinalValues = () =>
 export const getNodeLabelFunction = createDeepEqualSelector(
   protocolRegistry,
   variableRegistry => (node) => {
+    // Get the node entity section of the variable registry
     const nodeInfo = variableRegistry && variableRegistry.node;
+
+    // Get the display variable by looking up the node type in the variable registry
     const displayVariable = nodeInfo && node && node.type && nodeInfo[node.type] &&
       nodeInfo[node.type].displayVariable;
+
+    // For fallback: get the first variable of type 'text'
     const firstTextVariable = nodeInfo && node && node.type && nodeInfo[node.type] &&
       nodeInfo[node.type].variables && findKey(nodeInfo[node.type].variables, ['type', 'text']);
-    return node.label ||
-      (displayVariable && node[displayVariable]) ||
-      (firstTextVariable && node[firstTextVariable]) ||
-      String(node.id);
+
+    // Get the data model properties from the node
+    const nodeDataModelProps = node[NodeAttributesProperty];
+    // Try to return the label prop
+    // else try to use the displayVariable
+    // else try to use the first text variable
+    // else return the string "No label"
+    return nodeDataModelProps.label ||
+      (displayVariable && nodeDataModelProps[displayVariable]) ||
+      (firstTextVariable && nodeDataModelProps[firstTextVariable]) ||
+      'No label';
   },
 );
 
-export const makeNetworkNodesForSubject = () => {
+/**
+ * makeNetworkNodesForType()
+ * Get the current prompt/stage subject, and filter the network by this node type.
+*/
+
+export const makeNetworkNodesForType = () => {
   const getSubject = makeGetSubject();
   return createSelector(
     networkNodes, getSubject,
     (nodes, subject) => filter(nodes, ['type', subject.type]),
   );
 };
 
+/**
+ * makeNetworkNodesForPrompt
+ * Take the "additional attributes" specified by the current prompt, and filter nodes of the current
+ * prompt type
+*/
+
 export const makeNetworkNodesForPrompt = () => {
-  // used to check prompt ids
   const getAttributes = makeGetAdditionalAttributes();
-  const networkNodesForSubject = makeNetworkNodesForSubject();
+  const networkNodesForSubject = makeNetworkNodesForType();
 
   return createSelector(
     networkNodesForSubject, getAttributes,
-    (nodes, attributes) => filter(nodes, attributes),
+    (nodes, attributes) =>
+      filter(nodes, node => isMatch(node[NodeAttributesProperty], attributes)),
```

Comment:
No need to invoke isMatch here, filter uses that internally. Probably the matchesProperty is the syntax to go with here:

```
// `_.matchesProperty` iteratee shorthand.
filter(nodes, [NodeAttributesProperty, attributes])

// `_.matches` iteratee shorthand (deep equals)
filter(nodes, { [NodeAttributesProperty]: attributes })
```